icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +36,103 @@ def show_results(
+                  for each one of the losses.
+    """
+    device = torch.device("cpu")
+    model = model.to(device)
+    dl = valid_dl(dataset, batch_size=1, num_workers=0, shuffle=False)
+
+    torch.manual_seed(0)
+    samples_plus_losses = []
+    losses_stats = {
+        "loss_classifier": [],
+        "loss_box_reg": [],
+        "loss_objectness": [],
+        "loss_rpn_box_reg": [],
+        "loss_total": [],
+    }
+    for_text = ["loss_classifier", "loss_box_reg", "loss_total"]
+
+    with torch.no_grad():
+        for (x, y), sample in pbar(dl):
+            loss = model(x, y)
+            loss = {k: float(v.numpy()) for k, v in loss.items()}
+            loss["loss_total"] = sum(loss.values())
+
+            text = ""
+            for l in losses_stats.keys():
+                losses_stats[l].append(loss[l])
+                if l in for_text:
+                    text += f"{l}: {round(loss[l], 5)}\n"
+
+            text += f"IMG: {sample[0]['filepath'].name}"
+            loss["text"] = text
+
+            sample[0].update(loss)
+            samples_plus_losses.append(sample[0])
+
+    losses_stats = {k: get_stats(v) for k, v in losses_stats.items()}
+    return samples_plus_losses, losses_stats
+
+
+def plot_top_losses(
+    model: nn.Module,
+    dataset: Dataset,
+    sort_by: str = "loss_total",
+    n_samples: int = 5,
+) -> Tuple[List[dict], List[dict], dict]:
+    """
+    Gets a dataset and a model as input. Calculates losses for each sample in the dataset.
+    Sorts samples by `sort_by` (e.g. one of the losses). Runs model predictions on samples.
+    Plots the top sorted `n_samples` and returns losses + predictions + losses_stats.
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+    sort_by: (str) the loss to sort samples by
+    n_samples: how many samples to show
+
+    Returns
+    -------
+    sorted_samples: those are the samples (dictionaries) contained in the original dataset
+                    enriched with losses and sorted by `sort_by`.
+    sorted_preds: `model` predictions on `sorted_samples`. Same order as `sorted_samples`.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    model_inf = deepcopy(
+        model
+    )  # we need to create a copy of the model as `get_preds` puts it in eval mode and we need it in training mode
+    samples, losses_stats = get_losses(model, dataset)
+    _, preds = get_preds(model_inf, dataset)
+    sorted_samples, sorted_preds, annotations = sort_losses(samples, preds, by=sort_by)
+    assert len(sorted_samples) == len(samples) == len(preds) == len(sorted_preds)
+    show_preds(
+        samples=sorted_samples[:n_samples],
+        preds=sorted_preds[:n_samples],
+        annotations=annotations[:n_samples],
+    )
+
+    return sorted_samples, sorted_preds, losses_stats
```

Comment:
Can we break this function into two? One that calculates these statistics and another one to plot? (so the user might want to get the statistics without plotting)

Reply:
Actually it is already broken in 2, as I invoke `get_losses` inside `top_plot_losses`.
I am just returning `losses_stats` as a courtesy.
But this can be avoided, e.g. we simply don't return `losses_stats` and the user will have to manually invoke `get_losses` to get stats.
It seemed more work for nothing.
Wdyt?

Reply:
Hm... Makes sense.

What I was originally thinking is to have `plot_top_losses` receive as input the result of `get_losses`, but I think what you made makes more sense! Give me some time to think more about it :sweat_smile:

Reply:
@lgvaz any thoughts on this one?

Reply:
I think we leave it like this =)